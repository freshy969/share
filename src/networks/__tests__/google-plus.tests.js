import { when } from 'mobx';
import GooglePlus from '../google-plus';

const requestMock = () => {
  const req = {};
  req.post = jest.fn(() => req);
  req.set = jest.fn(() => req);
  req.send = jest.fn();

  return req;
};

describe('Share > GooglePlus', () => {
  test('url', () => {
    const googlePlus = GooglePlus.create({});
    googlePlus.entityLink = jest.fn();
    googlePlus.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauty-of-gullfoss');

    expect(googlePlus.url({ type: 'post', id: 60 })).toMatchSnapshot();
  });

  test('requestCount success', done => {
    const request = requestMock();
    const googlePlus = GooglePlus.create({}, { request });
    googlePlus.entityLink = jest.fn();
    googlePlus.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauty-of-gullfoss');

    request.send.mockResolvedValueOnce({
      body: {
        id: 'p',
        result: {
          kind: 'pos#plusones',
          id: 'https://demo.frontity.com/the-beauty-of-gullfoss',
          isSetByViewer: false,
          metadata: {
            type: 'URL',
            globalCounts: {
              count: 123,
            },
          },
          abtk: '',
        },
      },
    });

    when(
      () => googlePlus.count({ type: 'post', id: 60 }),
      () => {
        expect(googlePlus.count({ type: 'post', id: 60 })).toBe(123);
        expect(googlePlus.entityLink.mock.calls).toMatchSnapshot();
        expect(googlePlus.entityLink.mock.results).toMatchSnapshot();
        expect(request).toMatchSnapshot();
        done();
      },
    );

    googlePlus.requestCount({ type: 'post', id: 60 });
  });

  test('requestCount fails', async () => {
    const request = requestMock();

    const googlePlus = GooglePlus.create({}, { request });
    googlePlus.entityLink = jest.fn();
    googlePlus.entityLink.mockReturnValue('https://demo.frontity.com/the-beauty-of-gullfoss');

    // First request (request fails)
    request.send.mockRejectedValueOnce('fail');
    await googlePlus.requestCount({ type: 'post', id: 60 });
    expect(googlePlus.count({ type: 'post', id: 60 })).toBe(null);

    // Second request (bad response)
    request.send.mockResolvedValueOnce({ bad: 'response' });
    await googlePlus.requestCount({ type: 'post', id: 60 });
    expect(googlePlus.count({ type: 'post', id: 60 })).toBe(null);

    expect(request.send.mock.calls).toMatchSnapshot();
  });
});
