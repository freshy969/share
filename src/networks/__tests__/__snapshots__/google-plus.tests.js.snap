// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > GooglePlus requestCount fails 1`] = `
Array [
  Array [
    Object {
      "apiVersion": "v1",
      "id": "p",
      "jsonrpc": "2.0",
      "key": "p",
      "method": "pos.plusones.get",
      "params": Object {
        "groupId": "@self",
        "id": "https://demo.frontity.com/the-beauty-of-gullfoss",
        "nolog": true,
        "source": "widget",
        "userId": "@viewer",
      },
    },
  ],
  Array [
    Object {
      "apiVersion": "v1",
      "id": "p",
      "jsonrpc": "2.0",
      "key": "p",
      "method": "pos.plusones.get",
      "params": Object {
        "groupId": "@self",
        "id": "https://demo.frontity.com/the-beauty-of-gullfoss",
        "nolog": true,
        "source": "widget",
        "userId": "@viewer",
      },
    },
  ],
]
`;

exports[`Share > GooglePlus requestCount success 1`] = `
Array [
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
]
`;

exports[`Share > GooglePlus requestCount success 2`] = `
Array [
  Object {
    "isThrow": false,
    "value": "https://demo.frontity.com/the-beauty-of-gullfoss",
  },
]
`;

exports[`Share > GooglePlus requestCount success 3`] = `
Object {
  "post": [MockFunction] {
    "calls": Array [
      Array [
        "https://clients6.google.com/rpc",
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": [Circular],
      },
    ],
  },
  "send": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "apiVersion": "v1",
          "id": "p",
          "jsonrpc": "2.0",
          "key": "p",
          "method": "pos.plusones.get",
          "params": Object {
            "groupId": "@self",
            "id": "https://demo.frontity.com/the-beauty-of-gullfoss",
            "nolog": true,
            "source": "widget",
            "userId": "@viewer",
          },
        },
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": Promise {},
      },
    ],
  },
  "set": [MockFunction] {
    "calls": Array [
      Array [
        "Accept",
        "application/json",
      ],
      Array [
        "Content-type",
        "application/json;charset=UTF-8",
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": [Circular],
      },
      Object {
        "isThrow": false,
        "value": [Circular],
      },
    ],
  },
}
`;

exports[`Share > GooglePlus url 1`] = `"https://plus.google.com/share?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauty-of-gullfoss"`;

exports[`Share > GooglePlus url 2`] = `
Array [
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
]
`;

exports[`Share > GooglePlus url 3`] = `
Array [
  Object {
    "isThrow": false,
    "value": "https://demo.frontity.com/the-beauty-of-gullfoss",
  },
]
`;
